components/service-waiter/cmd/component.go

```diff
@@ -56,6 +63,26 @@ var componentCmd = &cobra.Command{
 	},
 }
 
+var experimentsClient experiments.Client
+var once sync.Once
+
+func shouldSkipComponentWaiter(ctx context.Context) bool {
+	once.Do(func() {
+		experimentsClient = experiments.NewClient()
+		if experimentsClient == nil {
+			log.Error("failed to create experiments client")
+		}
+	})
+	if experimentsClient == nil {
+		// skip if failed to craete client refer
+		// it should never reach here, in case it happens, always skip
+		return true
+	}
+	return experimentsClient.GetBoolValue(ctx, experiments.ServiceWaiterSkipComponentsFlag, true, experiments.Attributes{
```

Comment:
should not it be false by default? otherwise if it fails to get configuration first it will continue?

Reply:
default true in case we can't stop service-waiter when we want to do it.

PR need to change since it needs configcat's env vars

I will check in preview env to see its behavior and see if first config get is slow

Reply:
I don't think testing with preview envs will instil sense of security in me :) 

We should be sure that it is actual value from config cat. I don't believe there is a guarantee on it. There is a chance that we don't get a proper value and then we don't have a guarantee on deployment order.

Reply:
cc @geropl for the opinion on it

Reply:
IMO we should default to a special value ("NONE"?): If we see this value, we re-fetch until we either a) hit a timeout, or b) until we get a valid value for the feature flag. On timeout, we should be loud about it in the logs. The timeout should be picked that it does not drastically extend the overall max. waiting time for a pod to be ready.

I think this is the only way to get it working with a feature flag reliably (if ConfigCat works, which we have to assume anyway), and still have an automatic fallback in case it does not. The feature flag being on only acts as a short-cut, then.

Reply:
If we want to get the actual value (wait if configcat fetch failed etc), we can simply change default to `false -> not skip` then retry logic will make it wait again.

If we want to make sure even configcat is broken we want a way to unblock ourselves, default to `true -> skip` works. (But I think this change will be quite stable (I have added unit test that it don't configure wrong target image 🙈🥹). So perberly we can ignore this case.)

A special value like `NONE` should behavior the same with default false.

So good if I change it to default false? @akosyakov @geropl

Reply:
Agree with timeout. I wonder what is the reason of hiccups with configcat, proxy deployment?